pandas/core/frame.py

```diff
@@ -650,8 +650,11 @@ def _repr_html_(self):
             max_cols = get_option("display.max_columns")
             show_dimensions = get_option("display.show_dimensions")
 
-            return self.to_html(max_rows=max_rows, max_cols=max_cols,
-                                show_dimensions=show_dimensions, notebook=True)
+            formatter = fmt.DataFrameFormatter(
```

Comment:
why the change?

Reply:
`_repr_html_` just calls `to_html` with `notebook=True`, so need to avoid the FutureWarning in the notebook.

eventually i'd like to have `_repr_html_` instantiate `NotebookFormatter` directly and `to_html` instantiate `HTMLFormatter` directly. But at the moment neither are subclasses of `DataFrameFormatter`